x-pack/legacy/plugins/watcher/server/np_ready/types.ts

```diff
@@ -17,6 +17,6 @@ export interface ServerShim {
   };
 }
 
-export interface ServerShimWithRouter extends ServerShim {
+export interface NPServer {
```

Comment:
nit: There's no concept of "server" in NP, so using that as a name for the type and parameters is kinda legacy baggage. Maybe something like `RouteDependencies` or `RouteDeps`?